agent/agent-profiler/agent-service-profiler/src/main/java/com/microsoft/applicationinsights/serviceprofilerapi/profiler/JfrProfiler.java

```diff
@@ -133,13 +133,23 @@ public boolean initialize(
         flightRecorderConnection = FlightRecorderDiagnosticCommandConnection.connect(mbeanServer);
       }
     } catch (Exception e) {
-      LOGGER.error("Failed to connect to mbean", e);
+      String message =
+          "Unable to initialize the profiler."
+              + (isOpenJ9Jvm()
+                  ? " OpenJ9 JVM is not supported. Instead, please use an OpenJDK JVM."
+                  : "");
+      LOGGER.error(message, e);
```

Comment:
I'd suggest instead checking for OpenJ9 proactively before initializing the profiler

Reply:
Done; the drawback of doing this is that it won't become possible to benefit from the current initialization code if JFR is one day integrated into OpenJ9.